src/NLog/Config/LoggingConfiguration.cs

```diff
@@ -361,10 +369,12 @@ internal void Dump()
         internal void FlushAllTargets(AsyncContinuation asyncContinuation)
         {
             var uniqueTargets = new List<Target>();
-            foreach (var rule in this.LoggingRules)
+            for (int i = 0; i < this.LoggingRules.Count; i++)
```

Comment:
Do you think this needs to be fixed? If not can you resolve the conflicts, I can merge this.